addons/sale_stock/models/account_move.py

```diff
@@ -39,6 +39,7 @@ def _get_invoiced_lot_values(self):
 
         current_invoice_amls = self.invoice_line_ids.filtered(lambda aml: aml.display_type == 'product' and aml.product_id and aml.product_id.type in ('consu', 'product') and aml.quantity)
         all_invoices_amls = current_invoice_amls.sale_line_ids.invoice_lines.filtered(lambda aml: aml.move_id.state == 'posted').sorted(lambda aml: (aml.date, aml.move_name, aml.id))
+        all_invoices_amls = self._adjust_amls_for_invoiced_lot_valuation(all_invoices_amls)
```

Comment:
It would be better to write a `ensure_one` filter so you can use it in the `filtered` and overwrite it MX side

```suggestion
        all_invoices_amls = current_invoice_amls.sale_line_ids.invoice_lines.filtered(lambda aml: aml._your_filter()).sorted(lambda aml: (aml.date, aml.move_name, aml.id))
```